sklearn/preprocessing/_discretization.py

```diff
@@ -174,6 +208,36 @@ def fit(self, X, y=None):
                 " instead."
             )
 
+        n_samples, n_features = X.shape
+
+        if self.strategy == "quantile" and self.subsample is not None:
+            if self.subsample == "warn":
+                if n_samples > 2e5:
+                    warnings.warn(
+                        "In version 1.2 onwards, subsample=2e5 "
```

Comment:
The change should happen in 2 versions. The feature was developed probably for 1.0 at first.
```suggestion
                        "In version 1.3 onwards, subsample=2e5 "
```